docs/excel/custom-functions-authentication.md

```diff
@@ -85,19 +79,67 @@ function receiveTokenFromCustomFunction() {
 
 ## General guidance
 
-Office Add-ins are web-based and you can use any web authentication technique. There is no particular pattern or method you must follow to implement your own authentication with custom functions. You may wish to consult the documentation about various authentication patterns, starting with [this article about authorizing via external services](../develop/auth-external-add-ins.md).  
+Office Add-ins are web-based and you can use any web authentication technique. There is no particular pattern or method you must follow to implement your own authentication with UI-less custom functions. You may wish to consult the documentation about various authentication patterns, starting with [this article about authorizing via external services](../develop/auth-external-add-ins.md).  
 
 Avoid using the following locations to store data when developing custom functions:  
 
-- `localStorage`: Custom functions do not have access to the global `window` object and therefore have no access to data stored in `localStorage`.
+- `localStorage`: UI-less custom functions do not have access to the global `window` object and therefore have no access to data stored in `localStorage`.
 - `Office.context.document.settings`:  This location is not secure and information can be extracted by anyone using the add-in.
 
+## Dialog box API example
+
+In the following code sample, the function `getTokenViaDialog` uses the `Dialog` API's `displayWebDialogOptions` function to display a dialog box. This sample is provided only to show the capabilities of the `Dialog` object.
+
+```JavaScript
+/**
+ * Function retrieves a cached token or opens a dialog box if there is no saved token. Note that this is not a sufficient example of authentication but is intended to show the capabilities of the Dialog object.
+ * @param {string} url URL for a stored token.
+ */
+function getTokenViaDialog(url) {
+  return new Promise (function (resolve, reject) {
+    if (_dialogOpen) {
```

Comment:
What's up with the `_` before the variable names? Is that a standard in these APIs?

Reply:
It's a JS thing that can mean two things: it's private or an unused variable